tests/test_scalars.py

```diff
@@ -24,17 +39,644 @@
+    check(
+        assert_type(td - as_ndarray_td64, npt.NDArray[np.timedelta64]),
+        np.ndarray,
+        np.timedelta64,
+    )
+    check(assert_type(td - as_nat, NaTType), NaTType)
+    check(assert_type(as_period - td, pd.Period), pd.Period)
+    check(assert_type(as_timestamp - td, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_datetime - td, dt.datetime), dt.datetime)
+    check(assert_type(as_date - td, dt.date), dt.date)
+    check(assert_type(as_datetime64 - td, pd.Timestamp), pd.Timestamp)
+    # pyright can't know that as6 + td calls td.__rsub__(as6),  not timedelta.__sub__
+    # https://github.com/microsoft/pyright/issues/4088
+    check(
+        assert_type(
+            as_dt_timedelta - td,  # pyright: ignore[reportGeneralTypeIssues]
+            pd.Timedelta,
+        ),
+        pd.Timedelta,
+    )
+    check(assert_type(as_timedelta64 - td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(as_timedelta_index - td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(as_timedelta_series - td, TimedeltaSeries), pd.Series)
+    check(assert_type(as_period_index - td, pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(as_datetime_index - td, pd.DatetimeIndex), pd.DatetimeIndex)
+    # pyright is wrong here because ndarray.__sub__(Timedelta) is NotImplemented
+    check(
+        assert_type(
+            as_ndarray_td64 - td,  # pyright: ignore[reportGeneralTypeIssues]
+            npt.NDArray[np.timedelta64],
+        ),
+        np.ndarray,
+        np.timedelta64,
+    )
+    # pyright is wrong here because ndarray.__sub__(Timedelta) is NotImplemented
+    check(
+        assert_type(
+            as_ndarray_dt64 - td,  # pyright: ignore[reportGeneralTypeIssues]
+            npt.NDArray[np.datetime64],
+        ),
+        np.ndarray,
+        np.datetime64,
+    )
+    check(assert_type(as_nat - td, NaTType), NaTType)
+
+
+def test_timedelta_mul_div() -> None:
+    td = pd.Timedelta("1 day")
+
+    with pytest.warns(FutureWarning):
+        i_idx = cast(pd.Int64Index, pd.Index([1, 2, 3], dtype=int))
+        f_idx = cast(pd.Float64Index, pd.Index([1.2, 2.2, 3.4], dtype=float))
+
+    np_intp_arr: npt.NDArray[np.integer] = np.array([1, 2, 3])
+    np_float_arr: npt.NDArray[np.floating] = np.array([1.2, 2.2, 3.4])
+
+    md_int = 3
+    md_float = 3.5
+    md_ndarray_intp = np_intp_arr
+    md_ndarray_float = np_float_arr
+    mp_series_int = pd.Series([1, 2, 3], dtype=int)
+    md_series_float = pd.Series([1.2, 2.2, 3.4], dtype=float)
+    md_int64_index = i_idx
+    md_float_index = f_idx
+    md_timedelta_series = pd.Series(pd.timedelta_range("1 day", periods=3))
+
+    check(assert_type(td * md_int, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td * md_float, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td * md_ndarray_intp, np.ndarray), np.ndarray)
+    check(assert_type(td * md_ndarray_float, np.ndarray), np.ndarray)
```

Comment:
For the above two tests, should we do `npt.NDArray[np.integer]` and `npt.NDArray[np.float]` respectively in the `assert_type` test?

Reply:
These have been fixed. Both are timedelta64.